crates/bevy_ecs/src/result.rs

```diff
@@ -0,0 +1,10 @@
+//! Contains error and result helpers for use in fallible systems.
+
+/// A dynamic error type for use in fallible systems.
+pub type Error = Box<dyn core::error::Error + Send + Sync + 'static>;
+
+/// A result type for use in fallible systems.
+pub type Result<T = (), E = Error> = core::result::Result<T, E>;
```

Comment:
Since `IntoSystemConfigs` doesn't provide generic params, I'm not sure these type defaults make sense, unless there's a fast follow to support generic errors.

Reply:
Agreed, however I want to have these in the bevy prelude. If we put a `Result` without the generics in the prelude, it will prevent the normal result type from being used. By only adding defaults with an alias, I'm hoping we will be able to write systems that use the default `Result` and functions that use `Result<T, E>` in the same module without issue.

Reply:
There's a good precedent here: https://docs.rs/anyhow/latest/anyhow/type.Result.html

> `anyhow::Result` may be used with one or two type parameters.